core/src/test/java/com/linecorp/armeria/client/HttpResponseDecoderTest.java

```diff
@@ -107,4 +111,18 @@ void confirmResponseStartAndEndInTheSameThread(SessionProtocol protocol)
         latch.await(System.getenv("CI") != null ? 60 : 10, TimeUnit.SECONDS);
         assertThat(failed.get()).isFalse();
     }
+
+    @ParameterizedTest
+    @EnumSource(value = SessionProtocol.class, names = {"H1C", "H2C"})
+    void responseCompleteNormallyIfRequestIsAborted(SessionProtocol protocol) throws Exception {
+        final WebClient client = WebClient.of(server.uri(protocol));
+        final HttpRequestWriter request = HttpRequest.streaming(RequestHeaders.of(HttpMethod.POST, "/"));
+        final AggregatedHttpResponse res = client.execute(request).aggregate().join();
+        assertThat(res.contentUtf8()).isEqualTo("Hello, Armeria!");
+        // The request is aborted in HttpResponseDecoder.close(...) after the client receives the response.
```

Comment:
Abort sounds like an error.. I wonder if there's another way to say this as it sounds scary terminology in response to someone calling close (as opposed to cancel or abort)

```suggestion
        // The stream is aborted in HttpResponseDecoder.close(...) after the client receives the response.
```

Reply:
Indeed, we call `HttpRequest.abort()` here https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java#L299`

Reply:
Perhaps, we can check if it's an instance of `StreamWriter` and call `StreamWriter.close()` if it is. Let me try that.

Reply:
It's an `HeaderOverridingHttpRequest` so I think I have to refactor the `StreamMessage` a little bit to get it working.
Let me make an issue for that. 😄

Reply:
#3117 Created.